django/db/backends/utils.py

```diff
@@ -125,9 +125,10 @@ def typecast_time(s):  # does NOT store time zone information
     hour, minutes, seconds = s.split(':')
     if '.' in seconds:  # check whether seconds have a fractional part
         seconds, microseconds = seconds.split('.')
+        microseconds = microseconds.ljust(6, '0')
```

Comment:
This will blow up if any backend returns >6 digits of sub-second precision

MySQL can't do this, checking if SQLite can...